crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1406,14 +1407,63 @@ impl<'db> TypeInferenceBuilder<'db> {
         let ast::StmtAugAssign {
             range: _,
             target,
-            op: _,
+            op,
             value,
         } = assignment;
-        self.infer_expression(value);
-        self.infer_expression(target);
 
-        // TODO(dhruvmanila): Resolve the target type using the value type and the operator
-        Type::Todo
+        // Resolve the target type, assuming a load context.
+        let target_type = match &**target {
+            Expr::Name(name) => {
+                self.store_expression_type(target, Type::None);
+                self.infer_name_load(name)
+            }
+            Expr::Attribute(attr) => {
+                self.store_expression_type(target, Type::None);
+                self.infer_attribute_load(attr)
+            }
+            _ => self.infer_expression(target),
+        };
+        let value_type = self.infer_expression(value);
+
+        // TODO(charlie): Add remaining branches for different types of augmented assignments.
+        if let (Operator::Sub, Type::Instance(class)) = (*op, target_type) {
+            let class_member = class.class_member(self.db, "__isub__");
+            let call = class_member.call(self.db, &[value_type]);
```

Comment:
You need to explicitly pass in the `self` argument here because we've accessed the method on the class rather than the instance, so you're getting the original function object rather than a method object from the `class.class_member` call

```suggestion
            let call = class_member.call(self.db, &[target_type, value_type]);
```

This will matter when we start checking that function calls are being passed arguments of the correct types